README.md

```diff
@@ -4,33 +4,301 @@ sktm - skt manager
+[`example-project.yaml`](example-project.yaml) file, using Jenkins Job
+Builder:
+
+    jenkins-jobs update example-project.yaml
+
+Then go to the created project in Jenkins UI and click "[Build
+Now](http://localhost:8080/job/sktm/build)" to check that it works, and to
+have the pipeline parameters (which sktm requires) registered by Jenkins.
+
+Afterwards, every time you would change your YAML project description you will
+need to re-run `jenkins-jobs update example-project.yaml` and build the
+project manually once, as described above.
+
+### SKTM
+
+Create the `~/.sktmrc` file telling sktm how to access Jenkins:
+
+    cat >~/.sktmrc <<EOF
     [config]
-    jurl = http://jenkins.baseurl.com:8080
-    jlogin = mylogin
-    jpass = mypassword
-    jjname = jenkinsjobname
+    jurl = http://localhost:8080
+    jlogin = sktm
+    jpass = sesame
+    EOF
+
+Usage
+-----
+On the first execution sktm creates a database file (`~/.sktm.db` by default),
+which is then used to track status of tested kernel branches, patches, and
+the execution of Jenkins jobs sktm submits.
+
+### Establishing baseline
+
+Before beginning testing a kernel branch, you need to establish a working
+"baseline" commit, the patches would be applied on:
+
+    ./sktm.py -v --jjname <JENKINS_PROJECT> baseline <GIT_REPO_URL> <GIT_REF>
+
+Here, `<JENKINS_PROJECT>` would be the name of the Jenkins project running
+skt, `<GIT_REPO_URL>` would be a kernel Git repository URL, and `<GIT_REF>`
+would be a published Git reference in that repository from which to start
+locating a stable "baseline" commit, and usually will be a branch name.
+
+E.g., following our setup example above, this command would find a baseline
+commit in the current "scsi" tree:
+
+    ./sktm.py -v --jjname sktm baseline \
+              git://git.kernel.org/pub/scm/linux/kernel/git/mkp/scsi.git \
+              for-next
+
+And this would do the same for the "net-next" tree:
+
+    ./sktm.py -v --jjname sktm baseline \
+              git://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next.git \
+              master
+
+You will need to run the "baseline" command periodically to have your baseline
+commits up-to-date, to allow newer patches to apply.
+
+### Testing first patches
+
+Once your `baseline` command has finished, you will be able to start checking
+patches from a Patchwork instance. However, you would need to specify the
+patch to start from on the first run, and would need to use somewhat
+different commands for Patchwork v1 and v2 instances.
+
+For Patchwork v1 run:
+
+    ./sktm.py -v --jjname <JENKINS_PROJECT> patchwork \
+              <GIT_REPO_URL> \
+              <PATCHWORK_BASE_URL> <PATCHWORK_PROJECT> \
+              --lastpatch <PATCHWORK_PATCH_ID>
+
+and for Patchwork v2 run:
+
+    ./sktm.py -v --jjname <JENKINS_PROJECT> patchwork \
+              <GIT_REPO_URL> \
+              --restapi <PATCHWORK_BASE_URL> <PATCHWORK_PROJECT> \
+              --lastpatch <PATCHWORK_TIMESTAMP>
```

Comment:
Not related to the guide, the difference between passing patch ID and timestamp is irritable, unfriendly and needs to be fixed.

Reply:
Agreed 100%. Care to submit an issue?

Reply:
Submitted #46